pkg/shell/index.js

```diff
@@ -711,6 +760,12 @@ define([
                         perform_jump(control);
                     return;
 
+                } else if (control.command === "expect-restart") {
+                    /* watchdog handles current host for now */
+                    if (control.host != cockpit.transport.host)
```

Comment:
Shouldn't a null here mean that the default_host of the component which sent the message is the one that's restarting?